packages/runtime/src/useCases/consumption/attributes/DeletePeerSharedAttributeAndNotifyOwner.ts

```diff
@@ -0,0 +1,58 @@
+import { Result } from "@js-soft/ts-utils";
+import { AttributesController, ConsumptionIds, LocalAttribute, NotificationsController } from "@nmshd/consumption";
+import { Notification, PeerSharedAttributeDeletedByPeerNotificationItem } from "@nmshd/content";
+import { AccountController, CoreId, MessageController } from "@nmshd/transport";
+import { Inject } from "typescript-ioc";
+import { AttributeIdString, RuntimeErrors, SchemaRepository, SchemaValidator, UseCase } from "../../common";
+
+export interface DeletePeerSharedAttributeAndNotifyOwnerRequest {
+    attributeId: AttributeIdString;
+}
+
+class Validator extends SchemaValidator<DeletePeerSharedAttributeAndNotifyOwnerRequest> {
+    public constructor(@Inject schemaRepository: SchemaRepository) {
+        super(schemaRepository.getSchema("DeletePeerSharedAttributeAndNotifyOwnerRequest"));
+    }
+}
+
+export class DeletePeerSharedAttributeAndNotifyOwnerUseCase extends UseCase<DeletePeerSharedAttributeAndNotifyOwnerRequest, Notification> {
+    public constructor(
+        @Inject private readonly attributeController: AttributesController,
+        @Inject private readonly accountController: AccountController,
+        @Inject private readonly messageController: MessageController,
+        @Inject private readonly notificationsController: NotificationsController,
+        @Inject validator: Validator
+    ) {
+        super(validator);
+    }
+
+    protected async executeInternal(request: DeletePeerSharedAttributeAndNotifyOwnerRequest): Promise<Result<Notification>> {
+        const peerSharedAttributeId = CoreId.from(request.attributeId);
+        const peerSharedAttribute = await this.attributeController.getLocalAttribute(peerSharedAttributeId);
+
+        if (typeof peerSharedAttribute === "undefined") {
+            return Result.fail(RuntimeErrors.general.recordNotFound(LocalAttribute));
+        }
+
+        if (!peerSharedAttribute.isPeerSharedAttribute(peerSharedAttribute.shareInfo?.peer)) {
+            return Result.fail(RuntimeErrors.attributes.isNotPeerSharedAttribute(peerSharedAttributeId));
+        }
+
+        await this.attributeController.deleteAttribute(peerSharedAttribute);
```

Comment:
Reminder to not forget to delete all predecessors. :-)

Reply:
I already implemented it on another branch and thought I'd put it in a separate PR, so this one doesn't get even bigger.